storage/client/client.go

```diff
@@ -1315,6 +1315,42 @@ func (c *StorageClient) RuntimeTransactions(ctx context.Context, p apiTypes.GetE
 	return &ts, nil
 }
 
+// RuntimeTransaction returns a single runtime transaction.
+func (c *StorageClient) RuntimeTransaction(ctx context.Context, txHash string) (*RuntimeTransaction, error) {
+	cid, ok := ctx.Value(common.ChainIDContextKey).(string)
+	if !ok {
+		return nil, apiCommon.ErrBadChainID
+	}
+	runtime, ok := ctx.Value(common.RuntimeContextKey).(string)
+	if !ok {
+		return nil, apiCommon.ErrBadRuntime
+	}
+	qf := NewQueryFactory(cid, runtime)
+
+	t := RuntimeTransaction{}
+	err := c.db.QueryRow(
```

Comment:
so this reads the first matching one, right? is that by default the least recently inserted one?

Reply:
Answer before your comment:
`QueryRow()` throws an error if there's 0 or >1 result rows. So for ambiguous hashes, we'd return HTTP 500 right now. Same in consensus.

Now:
Still the same in consensus (I didn't touch it), but here I refactored so that we silently do a `LIMIT 1`, and yes, it's the most recent one that we return.